### PR TITLE
#2775: Created bar chart component for stats page and updated necessary dependencies

### LINKED ISSUE #2775: [Stats Page] - Bar Chart Component
### Description

We need a reusable bar chart component to display data. Using Chart.js, create a bar chart component that takes in x-axis data, y-axis data, axis labels, time frame, width, height, and graph title.

### Acceptance Criteria

- graph displays data in bar format
- hovering over a bar displays a tooltip with more detail
- all labels are properly formatted and sized/positioned
- width and height are responsive

### Proposed Solution

- new component file is created with suitable props

### Mocks

image
image

### PR DESCRIPTION
**Pull request by gcooper407**
## Changes

Created a Bar Chart component to be used on the Statistics page
Created a (temporary) sample bar chart on the stats page with sample data to showcase the components
Updated dependencies (package.json) to properly support the new components 

## Screenshots

Screenshot 2024-10-15 at 8 52 48 PM

Screenshot 2024-10-15 at 8 53 29 PM

Screenshot 2024-10-15 at 9 08 00 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2775

### CODE REVIEW

**Comment by wavehassman** [comment]
LGTM

File: src/frontend/src/components/StatsBarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        },
+        grid: {
+          color: '#6A6B6B'
+        }
+      }
+    }
+  };
+
+  return (
+    <div style={{ width: '100%', height: '100%', maxWidth: width, maxHeight: height }}>
```
Comment (walker-sean) [inline src/frontend/src/components/StatsBarChart.tsx]: Could you switch this over to a Box?

**Review by walker-sean — CHANGES_REQUESTED** [review]
This looks really cool!!

**Review by walker-sean — APPROVED** [review]
LGTM

**Pull request merged**